exporter/latency.go

```diff
@@ -30,9 +43,66 @@ func (e *Exporter) extractLatencyMetrics(ch chan<- prometheus.Metric, c redis.Co
 			var spikeLast, spikeDuration, max int64
 			if _, err := redis.Scan(latencyResult, &eventName, &spikeLast, &spikeDuration, &max); err == nil {
 				spikeDurationSeconds := float64(spikeDuration) / 1e3
-				e.registerConstMetricGauge(ch, "latency_spike_last", float64(spikeLast), eventName)
-				e.registerConstMetricGauge(ch, "latency_spike_duration_seconds", spikeDurationSeconds, eventName)
+				e.registerConstMetricGauge(outChan, "latency_spike_last", float64(spikeLast), eventName)
+				e.registerConstMetricGauge(outChan, "latency_spike_duration_seconds", spikeDurationSeconds, eventName)
 			}
 		}
 	}
 }
+
+func (e *Exporter) extractLatencyHistogramMetrics(outChan chan<- prometheus.Metric, infoAll string, redisConn redis.Conn) {
+	reply, err := redis.Values(doRedisCmd(redisConn, "LATENCY", "HISTOGRAM"))
+	if err != nil {
+		logHistogramErrOnce.Do(func() {
+			log.Errorf("WARNING, LOGGED ONCE ONLY: cmd LATENCY HISTOGRAM, err: %s", err)
+		})
+		log.Debugf("cmd LATENCY HISTOGRAM, err: %s", err)
+		return
+	}
+
+	for i := 0; i < len(reply); i += 2 {
+		cmd := string(reply[i].([]byte))
+		details, _ := redis.Values(reply[i+1], nil)
+
+		var totalCalls uint64
+		var bucketInfo []uint64
+
+		if _, err := redis.Scan(details, nil, &totalCalls, nil, &bucketInfo); err != nil {
+			break
```

Comment:
I'm not sure if this break can ever be reached unless there is a breaking change in Redis

Reply:
That's fine.